aten/src/ATen/native/quantized/QTensor.cpp

```diff
@@ -94,8 +94,10 @@ Tensor q_per_channel_zero_points_quant(const Tensor& self) {
       self.options().dtype(at::kLong));
 }
 
-Quantizer* quantizer(const Tensor& self) {
-  return get_qtensorimpl(self)->quantizer().get();
+IntArrayRef q_per_channel_axis_quant(const Tensor& self) {
```

Comment:
This is the very first occurrence of an `IntArrayRef` at return site of a native function, and I don't like it. The returned ref is non-owning. So what's its lifetime tied to? This is not documented anywhere. (In fact, the lifetime is tied to the *quantizer*, not the tensor itself; so the semantics don't even match `sizes()` and `strides()`, whose lifetime is tied to the tensor). Furthermore, this would have to be supported properly in the JIT, as all values in the stack must be owning. C.f. implementation of `sizes()`:

```
    Operator(
        "aten::size(Tensor self) -> int[]",
        [](Stack& stack) {
          RECORD_FUNCTION("size", last(stack, 1));
            
          auto t = std::move(pop(stack)).toTensor();
          pack(stack, t.sizes().vec());
          return 0;
        },    
        aliasAnalysisFromSchema()),
```

I suppose, from first principles, the ability to return a primitive array from a function is desirable functionality, I think we should think carefully about how we want to design this. As a first proposal, the return type of this function should be a `std::vector<>`, not an `IntArrayRef`.

cc @gchanan @zdevito